### PR TITLE
[SPARK-29752][SQL][TEST] make AdaptiveQueryExecSuite more robust

### PR DESCRIPTION
**Pull request by cloud-fan**
### What changes were proposed in this pull request?

instead of checking the exact number of local shuffle readers, we should check whether the number of shuffles is equal to the number of local readers.

### Why are the changes needed?

AQE is known to have randomness. We may pick different build side for broadcast join depending on which query stage finishes first. The decision to build side may add/remove shuffles downstream, so it's flaky to check the exact number of local shuffle readers.

### Does this PR introduce any user-facing change?

no

### How was this patch tested?

test only PR.

### CODE REVIEW

**Comment by cloud-fan** [comment]
cc @maryannxue @JkSelf

File: sql/core/src/test/scala/org/apache/spark/sql/execution/adaptive/AdaptiveQueryExecSuite.scala [hunk 1]
```diff
@@ -93,24 +99,7 @@ class AdaptiveQueryExecSuite
       assert(smj.size == 1)
       val bhj = findTopLevelBroadcastHashJoin(adaptivePlan)
       assert(bhj.size == 1)
-      checkNumLocalShuffleReaders(adaptivePlan, 2)
-    }
-  }
-
-  test("Change merge join to broadcast join and reduce number of shuffle partitions") {
```
Comment (cloud-fan) [inline sql/core/src/test/scala/org/apache/spark/sql/execution/adaptive/AdaptiveQueryExecSuite.scala]: This test case is duplicated with the above one now. We add local reader to both sides of broadcast join and won't apply "reduce number of shuffle partitions".

**Comment by cloud-fan** [comment]
retest this please

**Review by gatorsmile — APPROVED** [review]
LGTM

Thanks! Merged to master.

**Pull request closed without merging**